xt/lib/PageObject/App/Invoice.pm

```diff
@@ -21,10 +21,10 @@ sub _verify {
 }
 
 sub update {
-    my ($self) = @_;
+    my ($self,$elem) = @_;
 
     $self->find("*button", text => "Update")->click;
-    $self->session->page->body->maindiv->wait_for_content;
+    $self->session->page->body->maindiv->wait_for_content(replaces => $elem);
```

Comment:
I understand what you're doing here, but the "Update" button could serve the same purpose, no?

Reply:
Yes, if the whole page changes. What if only partial updates are done? Only the Customer/Vendor field seems to be updated.

Reply:
Indeed. Simplified